src/agent/command_handler/include/command_handler.hpp

```diff
@@ -91,6 +91,15 @@ namespace command_handler
 
                 if (cmd.value().ExecutionMode == module_command::CommandExecutionMode::SYNC)
                 {
+                    if (cmd.value().Module == "restart")
+                    {
+                        // This block is necessary to avoid potential race conditions between the restart and reporting
+                        // the command status. It allows detecting when the restart completes by setting the command to
+                        // IN_PROGRESS.
+                        cmd.value().ExecutionResult.ErrorCode = module_command::Status::IN_PROGRESS;
```

Comment:
I think at this point, the cmd is already in `IN_PROGRESS` status. It's created with this status when it is extracted from the queue.